### PR TITLE
Add CI pipeline

### LINKED ISSUE #173: Add GitHub Acitions to project
- Add analyze SonarCloud
- Run tests and report results

### PR DESCRIPTION
**Pull request by pro100user19**
- Add analyze SonarCloud
- Run tests and report results

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      - develop
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  build:
+    name: Check SonarCloud and start Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```
Comment (dshma) [inline .github/workflows/build.yml]: v3 could be used

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        with:
+          fetch-depth: 0
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
```
Comment (dshma) [inline .github/workflows/build.yml]: v3 could be used, take 'temurin' distribution
https://github.com/actions/setup-java

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        with:
+          java-version: 11
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Cache Gradle packages
+        uses: actions/cache@v1
```
Comment (dshma) [inline .github/workflows/build.yml]: using setup-java@v3 it would look like this:
cache: 'gradle'

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
```
Comment (dshma) [inline .github/workflows/build.yml]: v3 could be used

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Cache Gradle packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Start MongoDB v 5.0
+        uses: supercharge/mongodb-github-action@1.7.0
```
Comment (dshma) [inline .github/workflows/build.yml]: I think component scanning within test classes should be restricted to not checking the MongoConfig as this forces to start a MongoDB server when it is not even needed

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        with:
+          mongodb-version: 5.0
+          mongodb-username: supercharge
+          mongodb-password: secret
+          mongodb-db: schedules
+      - name: Start Redis v 5
+        uses: superchargejs/redis-github-action@1.1.0
```
Comment (dshma) [inline .github/workflows/build.yml]: same as for MongoDB

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        with:
+          redis-version: 5
+      - name: SonarCloud analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: ./gradlew sonarqube -x test --info
```
Comment (dshma) [inline .github/workflows/build.yml]: why info logging level?

**Review by dshma — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 7 above)
Comment (pro100user19) [reply .github/workflows/build.yml]: I looked through the logs and forgot to delete

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+          cache: 'gradle'
+      - run: ./gradlew build --no-daemon
```
Comment (dshma) [inline .github/workflows/build.yml]: why running the build two times?
lines 23 and 36

Comment (dshma) [inline .github/workflows/build.yml]: why --no-daemon?

File: .github/workflows/build.yml [hunk 9]
```diff
@@ -19,8 +19,6 @@ jobs:
         with:
           distribution: 'temurin'
           java-version: '11'
-          cache: 'gradle'
```
Comment (dshma) [inline .github/workflows/build.yml]: why removing the gradle packages cache?

**Review by dshma — APPROVED** [review]

**Pull request merged**